src/app/tracing/util/map-utils.ts

```diff
@@ -29,52 +30,78 @@ export interface RectConfig {
 const LAYER_ID_KEY = "layerId";
 const MAP_LAYER_ID = "MapLayer";
 
-const MAP_SOURCE: Map<MapType, () => OSM> = new Map([
-    [MapType.MAPNIK, () => new OSM()],
+const MAP_SOURCE: Map<TileServer, () => OSM> = new Map([
+    [TileServer.MAPNIK, () => new OSM()],
     // the following code is commented because
-    // the Black & White Map might be deactivatd only temporaryly
-    // ,
-    // [MapType.BLACK_AND_WHITE, () => new OSM({
+    // the Black & White Map might be deactivatd only temporarily
+    // [TileServer.BLACK_AND_WHITE, () => new OSM({
     //     url: 'https://tiles.wmflabs.org/bw-mapnik/{z}/{x}/{y}.png',
     //     crossOrigin: null
     // })]
 ]);
 
+//* ***** Question START: Is this code still needed? It doesn't seem to be in use anywhere. ***** */
+
 // the following code is commented because
-// the Black & White Map might be deactivatd only temporaryly
-// const availableMapTypes: MapType[] = [ MapType.MAPNIK, MapType.BLACK_AND_WHITE ];
-const availableMapTypes: MapType[] = [MapType.MAPNIK];
+// the Black & White Map might be deactivatd only temporarily
+const availableMapTypes: TileServer[] = [
+    TileServer.MAPNIK,
+    /* TileServer.BLACK_AND_WHITE, */
+];
 
-export function getAvailableMapTypes(): MapType[] {
+export function getAvailableMapTypes(): TileServer[] {
     return availableMapTypes;
 }
 
+//* ***** Question END: ***** */
+
 export function createOpenLayerMap(
-    mapConfig: MapConfigWithOptLayout,
+    mapConfig: MapViewConfig,
     target?: HTMLElement,
 ): ol.Map {
     const map = new ol.Map({
         target: target,
-        layers: [createMapLayer(mapConfig)],
+        layers: createMapLayer(mapConfig),
         controls: [],
     });
     return map;
 }
 
-function createMapLayer(mapConfig: MapConfigWithOptLayout): BaseLayer {
+function createMapLayer(mapConfig: MapViewConfig): Array<BaseLayer> {
+    const { mapType } = mapConfig;
+
+    // create a multi-layer map if both layers are present
+    if (mapType === MapType.TILES_AND_SHAPE) {
+        const topLayer = createShapeFileLayer(
+            mapConfig as NotNullish<ShapeFileSettings>,
+        );
+        topLayer.set(LAYER_ID_KEY, MAP_LAYER_ID, true);
```

Comment:
Why are you using for both layers the identical layer id? I used them for identification.